htdocs/accountancy/class/accountingaccount.class.php

```diff
@@ -389,6 +389,56 @@ public function checkUsage()
 		}
 	}
 
+    /**
+     * Find centralist accounting account of a subledger
+     *
+     * @param	string  $subledgeraccount   Subledger account
+     * @return  string                      Centralist accounting acoount of subledger account
+     */
+    public function searchAccountSubledgerInfo($subledgeraccount)
+    {
+        global $conf, $langs;
+
+        // Search on table
+        // Customers
+        $sql  = "(SELECT code_compta as type_customer FROM " . MAIN_DB_PREFIX . "societe";
+        $sql .= " WHERE code_compta LIKE '" . $this->db->escape($subledgeraccount) . "')";
```

Comment:
Are you sure you don't miss '%' into the like ?
If yes, you can use instead of the like, the  natural_search("code_compta", $subledgeraccount) to forge the "where"